src/main/java/io/javalin/core/JavalinConfig.java

```diff
@@ -188,6 +190,20 @@ public static void applyUserConfig(Javalin app, JavalinConfig config, Consumer<J
             config.inner.compressionStrategy = CompressionStrategy.NONE;
         }
 
+        //Apply compression strategy to all static file handlers, then start the handlers
+        if(config.inner.resourceHandler != null) {
```

Comment:
Can you do this inside `JettyResourceHandler` instead?

Reply:
I could potentially. The biggest issue is I need a guarantee that the Javalin user config has already been applied when the handlers are being initialized, so I can set the compression strategy on the handlers before they are started.

Since both compression strategy and static handlers are set during the same call to userConfig.accept, I am not sure how the order of their initialization would be determined.

Reply:
I did some more testing here, and it's exactly as I suspected. When the static handler is initialized via the addStaticFiles config method, the state of the compressionStrategy variable depends solely on the order in which the 2 options were called in user config. 

So this would only work correctly if you set compression strategy before adding the static file handlers, like so: 

```
    val app = Javalin.create {config ->
        config.compressionStrategy( CompressionStrategy(Brotli(), Gzip()) )
        config.addStaticFiles("/public")
    }.start(7070)
```

That's a big no no because it would require config options to be specified in the correct order.

Unless you know of a way to get around this problem, I don't see how I can avoid initializing the static handlers within JavalinConfig#applyUserConfig. As far as  I can tell, this is the only way to **guarantee** that user config has already been applied before passing it on to the handlers.

But as always, input is welcome! It's how I learn to do stuff in better ways :)